test/e2e_node/hugepages_test.go

```diff
@@ -85,7 +85,14 @@ func makePodToVerifyHugePages(baseName string, hugePagesLimit resource.Quantity)
 
 // configureHugePages attempts to allocate 10Mi of 2Mi hugepages for testing purposes
 func configureHugePages() error {
-	err := exec.Command("/bin/sh", "-c", "echo 5 > /proc/sys/vm/nr_hugepages").Run()
+	// Drop all kernel caches to make sure we can get the requested amount of huge pages
```

Comment:
Could you please expand on how dropping the kernel caches is necessary to ensure we get the requested amount of huge pages?

Reply:
One thing I forgot to say... thank you for linking the kernel docs! That is helpful :)

Reply:
Still curious about the answer to the specific q in my first comment when you get a chance :) Thanks!

Reply:
Hi,

After some more research, it looks like running memory compaction is more reasonable.

This is the docs from kernel.org about `/proc/sys/vm/compact_memory`:

```
Available only when CONFIG_COMPACTION is set. When 1 is written to the file,
all zones are compacted such that free memory is available in contiguous
blocks where possible. This can be important for example in the allocation of
huge pages although processes will also directly compact memory as required.
```

I have updated the PR to use that instead. Thoughts?